views/graph.py

```diff
@@ -363,16 +485,39 @@ def add_objects(self, layer, features):
 
         # get attribute information from selected layers
         items = []
+        existing_items = ["%s_%s"%(item.object_type.value,
+                                   str(item.object_id.value))
+                for item in self.model.rows]
         for feature in features:
-            item = {
-                'object_type': layer.name(),
-                'object_id': feature['id'],
-                'object_name': feature['display_name'],
-                'file_path': filename
-            }
-            items.append(item)
+            #check if object not already exist
+
+            if (layer.name() + '_' + str(feature['id'])) not in existing_items:
+                item = {
+                    'object_type': layer.name(),
+                    'object_id': feature['id'],
+                    'object_name': feature['display_name'],
+                    'file_path': filename
+                }
+                items.append(item)
+
+        if len(items) > 20:
+            msg = "%i nieuwe objecten zijn geselecteerd. Toevoegen aan de " \
+                  "grafiek kan enkele tijd duren. Wilt u doorgaan?"%len(items)

```

Comment:
Vreemd dat pyCharm deze niet zag, toch eens de instellingen controleren (of het script eens draaien)